[clarissatjx] iP

clarissatjx opened the pull request:
### Edith.java

> "Caw caw!" – Crow ([source](https://www.youtube.com/watch?v=-GTq_XazZPM))

Edith frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use


All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

clarissatjx committed:
Level-0

clarissatjx committed:
Level-0

clarissatjx committed:
Level-1

clarissatjx committed:
Level-2: Adding to a list

clarissatjx committed:
Level-3: marking and unmarking a task as done. new class: Task

clarissatjx committed:
Level-4: different types of tasks enabled

clarissatjx committed:
A-TextUiTesting: using automated testing

clarissatjx committed:
Level-5: handling errors, creating exceptions

clarissatjx committed:
Level-6: allows users to delete tasks from their todo list

clarissatjx committed:
Allow saving to do list to user's computer

clarissatjx committed:
Merge branch 'branch-Level-7': Add the ability to save the to do list to user's computer

clarissatjx committed:
Bot can now interpret date and time input

clarissatjx committed:
Merge branch 'branch-Level-8': Bot can now interpret date and time input

clarissatjx committed:
Add new class Storage.java that deals with loading tasks from the file and saving tasks in the file

clarissatjx committed:
Add Parser.java class which deals with making sense of the user command

clarissatjx committed:
Add Ui.java class which deals with interactions with the user

clarissatjx committed:
Add minor changes

clarissatjx committed:
Merge branch 'branch-A-MoreOOP': Refactor the code to extract out closely related code as classes.

clarissatjx committed:
Organize the classes into suitable java packages

clarissatjx committed:
Organize the classes into suitable java packages

clarissatjx committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

clarissatjx committed:
Add JUnit tests to test the behavior of the code

clarissatjx committed:
Merge branch 'branch-A-JUnit'

clarissatjx committed:
Edit build.gradle's entry point and jar-to-be-created's name

clarissatjx committed:
JavaDoc comments for all exceptions and tasks classes

clarissatjx committed:
Add JavaDocs to Edith, Parser, Storage, ToDoList and Ui

clarissatjx committed:
Merge branch 'branch-A-JavaDoc'

clarissatjx committed:
Modify code to observe coding standards

clarissatjx committed:
Merge branch-A-CodingStandard, resolve merge conflicts

clarissatjx committed:
Add minor changes to Ui

clarissatjx committed:
Add methods that allow user to find tasks based on keyword(s)

clarissatjx committed:
Fix merge conflicts

chenle228 commented on src/main/java/Edith.java:
```diff
@@ -10,16 +10,27 @@ public static void main(String[] args) {
         String farewell = " bye!! see you soon love <3";
```
I really love how you customise/personalise your bot! So cute:)

chenle228 commented on src/main/java/ToDo.java:
```diff
@@ -1,17 +1,29 @@
+            str = str + (i + 1) + "." + todoList.get(i).toString() + "\n";
         }
         return str;
     }
```
Good naming of methods and variables that follow the code quality guidelines!:)

chenle228 commented on src/main/java/EventTask.java:
```diff
@@ -0,0 +1,18 @@
+public class EventTask extends Task {
```
Good practice of OOP!

chenle228 commented on src/main/java/MissingEventDurationException.java:
```diff
@@ -0,0 +1,6 @@
+        super(" oops! event duration cannot be empty! please enter a duration after the task type and task name." +
+                "for example:\n      event cs2101 project meeting /from 4pm /to 7pm");
+    }
+}
```
I really like how you specified different types of Exception and refactored them into different classes when the instructions didn't explicitly ask for it! It really demonstrated how you kept OOP principle in mind when designing your solution:)

chenle228 reviewed (commented):
Overall I found your code very well-thought-out and well-written! Keep up the good work:)

clarissatjx committed:
Edit styling

The0nlyJuan commented on src/main/java/edith/Parser.java:
```diff
@@ -0,0 +1,149 @@
+        try {
+            List<String> userInputs = Arrays.asList(taskDuration.split(" /to "));
+            return userInputs.get(0);
+        } catch (IndexOutOfBoundsException e) {
```
There are methods in your code that are almost identical but with minor differences. For example, getTaskDeadline, getTaskDuration, getTaskStart, and getTaskEnd all have similar try-catch blocks. You could create a helper method that extracts parts of a string based on a delimiter, reducing duplication

The0nlyJuan reviewed (commented):
Some code duplications

ashea10 commented on src/main/java/edith/task/DeadlineTask.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Override
+    public String toString() {
+        if (this.getCompletionStatus()) {
+            return "[D][X] " + this.getTaskName() + " (by: " + this.deadline + ")";
+        } else {
+            return "[D][ ] " + this.getTaskName() + " (by: " + this.deadline + ")";
+        }
```
This part could perhaps be made less verbose by doing something like:
``` Java
return "[D][" + (this.getCompletionStatus() ? "X" : " ") + "] " + this.getTaskName() + " (by: " + this.deadline + ")";
```
Just a minor aesthetic issue which could reduce repetition.
Noticed this in a few other places as well.

ashea10 reviewed (commented):
LGTM, largely correct in terms of coding style, good job! Avoided many of the coding standard violations!

clarissatjx committed:
Add a GUI to the chatbot using javafx

clarissatjx committed:
Merge branch-Level-10

clarissatjx committed:
Change GUI

clarissatjx committed:
Use assert feature and edit javadocs comments

Assertions are currently not used.

Assert feature document important assumptions that should hold at
various points in the code. Assertions are suitable for verifying
assumptions about Internal Invariants, Control-Flow Invariants,
Preconditions, Postconditions, and Class Invariants. An assertion
failure indicates the programmer made a mistake in the code.

Assertions are now added to Ui.java only as most of the checking
needs to be done in this class.

JavaDocs comments were edited as many return types of methods were
changed in Ui.java

clarissatjx committed:
Merge pull request #1 from clarissatjx/branch-A-Assertions

Use assert feature and edit javadocs comments

clarissatjx committed:
Add enums to improve code quality

clarissatjx committed:
Improve code quality by deleting dead code

clarissatjx committed:
Merge pull request #2 from clarissatjx/branch-A-CodeQuality

Improve code quality

clarissatjx committed:
Add new function to edith: expense tracking

Edith bot can only track tasks and has pretty messy code.

Added new classes to accommodate for this new function.
Separated classes into more packages for easier reading
and hierachy. Packages are now split into 'task' and
'expense' and each package has a package 'exception' for
their respective exceptions.

clarissatjx committed:
Update README.md, fix bugs, add window name

clarissatjx committed:
Fix README.md

clarissatjx committed:
Fix README.md

clarissatjx committed:
Update README.md

clarissatjx committed:
Merge pull request #4 from clarissatjx/clarissatjx-patch-1

Update README.md

clarissatjx committed:
Fix minor bugs

clarissatjx committed:
Edit readme file to match
